sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessEvaluationContext.java

```diff
@@ -368,23 +368,22 @@ public CounterSet getCounters() {
    * {@link InProcessPipelineOptions#isShutdownUnboundedProducersWithMaxWatermark()}.
    */
   public boolean isDone(AppliedPTransform<?, ?, ?> transform) {
-    boolean done = true;
+    if (!watermarkManager
+        .getWatermarks(transform)
+        .getOutputWatermark()
+        .equals(BoundedWindow.TIMESTAMP_MAX_VALUE)) {
```

Comment:
Switched this to isBefore, which is much easier to read in this order.

NPEs should never occur from the watermark manager, as all of the watermarks at creation time. Switching the order decreases readability and also isn't actually particularly good protection against NPEs in this context.